libs/langgraph/src/pregel/runner.ts

```diff
@@ -77,31 +87,38 @@ export class PregelRunner {
    * @param options - Options for the execution.
    */
   async tick(options: TickOptions = {}) {
-    const { timeout, signal, retryPolicy, onStepWrite, maxConcurrency } =
-      options;
+    const { timeout, retryPolicy, onStepWrite, maxConcurrency } = options;
 
     const nodeErrors: Set<Error> = new Set();
     let graphBubbleUp: GraphBubbleUp | undefined;
+    const exceptionSignalController = new AbortController();
 
     // Start task execution
     const pendingTasks = Object.values(this.loop.tasks).filter(
       (t) => t.writes.length === 0
     );
 
+    const currentSignals = this._initializeAbortSignals({
+      exceptionSignalController,
+      timeout,
+      signal: options.signal,
+    });
+
     const taskStream = this._executeTasksWithRetry(pendingTasks, {
-      stepTimeout: timeout,
-      signal,
+      signals: currentSignals,
       retryPolicy,
       maxConcurrency,
     });
 
-    for await (const { task, error } of taskStream) {
+    for await (const { task, error, signalAborted } of taskStream) {
       this._commit(task, error);
       if (isGraphInterrupt(error)) {
         graphBubbleUp = error;
       } else if (isGraphBubbleUp(error) && !isGraphInterrupt(graphBubbleUp)) {
         graphBubbleUp = error;
-      } else if (error) {
+      } else if (error && (nodeErrors.size === 0 || !signalAborted)) {
+        // trigger task cancellation due to error
+        exceptionSignalController.abort();
         nodeErrors.add(error);
       }
```

Comment:
The goal here is to capture the exception that causes us to abort. In theory it's possible for multiple nodes to throw, so this also handles the edge case of capturing concurrent exceptions thrown before the node saw an abort.

Also if an external abort triggers a node to throw an `AbortError` or similar, we want to propagate that back out to the caller, so we always grab the first non-bubble-up error here, even if it occurred after the signal was aborted. Otherwise in the event that this happens as a result of some other error being thrown, we don't bother keeping the `AbortError` or whatever, as it's not causal and therefore unimportant to the caller.